spec/fastsync/vdd/fastsyncL2.tla

```diff
@@ -0,0 +1,679 @@
+    
+    
+    \/  /\ peersState' = pState
+        /\ inMsg' \in [type: {"blockResponse"}, peerId: FAULTY, block: Blocks]    
+
+CreateResponse(pState) == 
+    \/  SendBlockResponseMessage(pState)
+    \/  GrowBlockchain(pState)
+    \/  SendControlMessage(pState)
+                         
+    
+\* Peers consume a message and update it's local state. It then makes a single step, i.e., it sends at most single message.
+\* Message sent could be either a response to a request or faulty message (sent by faulty processes). 
+NextPeers ==
+    LET pState == HandleRequest(outMsg, peersState) IN
+    
+    \/  /\ outMsg' = NoMsg                    \* correct peers respond
+        /\ CreateResponse(pState) 
+    
+
+\* the composition of the node, the peers, the network and scheduler
+Init ==
+    /\ InitNode
+    /\ InitPeers
+    /\ turn = "Peers"
+    /\ inMsg = NoMsg
+    /\ outMsg = [type |-> "statusRequest", peerIds |-> blockPool.peerIds]
+    /\ nonDetFlag \in BOOLEAN
+        
+Next ==
+    IF turn = "Peers"
+    THEN
+        /\ NextPeers
+        /\ turn' = "Node"
+        /\ nonDetFlag' \in BOOLEAN
+        /\ UNCHANGED nvars
+    ELSE
+        /\ NextNode
+        /\ turn' = "Peers"
+        /\ nonDetFlag' \in BOOLEAN
+        /\ UNCHANGED peersState  
+
+
+
+FlipTurn == 
+ turn' = (
+  IF turn = "Peers" THEN 
+   "Node" 
+  ELSE
+   "Peers"
+ ) 
+        
+ 
+
+\* properties to check
+TypeOK ==
+    /\ state \in States
+    /\ inMsg \in InMsgs 
+    /\ outMsg \in OutMsgs
+    /\ turn \in {"Peers", "Node"}
+    /\ peersState \in [
+         peerHeights: [AllPeerIds -> Heights \union {NilHeight}],
+         statusRequested: SUBSET AllPeerIds,
+         blocksRequested: SUBSET [type: {"blockResponse"}, peerId: AllPeerIds, block: Blocks],
+         turn: {"status", "block"}
+        ]
+    /\ blockPool \in [
+                height: Heights,
+                peerIds: SUBSET AllPeerIds, 
+                peerHeights: [AllPeerIds -> Heights \union {NilHeight}],     
+                blockStore: [HeightsPlus -> Blocks \union {NilBlock}],
+                receivedBlocks: [Heights -> AllPeerIds \union {NilPeer}],
+                pendingBlocks: [Heights -> AllPeerIds \union {NilPeer}]
+           ] 
+           
+              
+Correctness1 == state = "finished" => 
+    \/ blockPool.peerIds = {} 
+    \/ blockPool.height >= MaxPeerHeight(blockPool)
+
+
+Correctness2 == 
+   \A p \in CORRECT: 
+        \/ p \notin blockPool.peerIds 
+        \/ [] (state = "finished" => blockPool.height >= blockPool.peerHeights[p] - 1)
+          
+
+Termination == WF_turn(FlipTurn) => <>(state = "finished") 
+     
+\* a few simple properties that trigger counterexamples
+
+\* Shows execution in which peer set is empty
+PeerSetIsNeverEmpty == blockPool.peerIds /= {}
+
+\* Shows execution in which state = "finished" and MaxPeerHeight is not equal to 1
+StateNotFinished == 
+    state /= "finished" \/ MaxPeerHeight(blockPool) = 1
+        
```

Comment:
I tried to run this and it failed with
```
TLC threw an unexpected exception.
This was probably caused by an error in the spec or model.
See the User Output or TLC Console for clues to what happened.
The exception was a java.lang.RuntimeException
: Attempted to enumerate S \ T when S:
TRUE
is not enumerable.
TLC was unable to fingerprint.

Fingerprint Stack Trace:
0) line 516, col 33 to line 516, col 63 of module fastyncL2
```

Could you add some information in README.md with intializations that worked for you? I tried with empty FAULTY and 2 correct peers, height 3

Reply:
Should work now.